### PR TITLE
Allow for extra image in metadata and to move image preview out of traits

### PR DESCRIPTION
**Pull request by newbreedofgeek**
To fix:

https://github.com/Itheum/sdk-mx-data-nft/issues/126

https://github.com/Itheum/sdk-mx-data-nft/issues/125

### CODE REVIEW

**Comment by damienen** [comment]
Preview can be moved out of traits, but it's debatable if it should be moved as it allows people to quickly see the preview from the marketplace.

The bonus image trait is redundant as we should just use the MultiversX NFT standard
Screenshot_20240406-145302
That would mean actively using the assets URIs. Those can be however many we want, more cam be updated to make the NFTs upgradable, etc.

**Comment by damienen** [comment]
Here's some documentation on it:
https://docs.multiversx.com/tokens/nft-tokens/#add-uris-to-nft

https://docs.multiversx.com/tokens/nft-tokens/#creation-of-an-nft see URIs at end of NFT creation

**Comment by newbreedofgeek** [comment]
that's fine, but we are going to aim to build this to be multi-chain compatible so we can bridge data nfts

File: src/common/mint-utils.ts [hunk 1]
```diff
@@ -63,8 +63,9 @@ export function createIpfsMetadata(
   dataNFTStreamPreviewUrl: string,
   address: string
 ) {
-  const metadata = {
+  const metadata: Record<string, any> = {
     description: `${datasetTitle} : ${datasetDescription}`,
+    data_preview_url: dataNFTStreamPreviewUrl,
     attributes: [] as object[]
   };
   const attributes = traits
```
[the remark this answers is not in the document]
Comment (damienen) [reply src/common/mint-utils.ts:71]: Cool, will add

[the remark this answers is not in the document]
Comment (damienen) [reply src/common/mint-utils.ts:71]: Added

File: src/common/mint-utils.ts [hunk 2]
```diff
@@ -61,11 +61,13 @@ export function createIpfsMetadata(
   datasetTitle: string,
   datasetDescription: string,
   dataNFTStreamPreviewUrl: string,
-  address: string
+  address: string,
+  extraAssets: string[]
```
Comment (newbreedofgeek) [inline src/common/mint-utils.ts:65]: Only add extraAssets to the NFT Metadata IF it's provided by the user, i.e. as it's optional and cal be empty... if it's empty we should not add it to NFT metadata file.

I.e. do an : if (extraAssets && extraAssets.length > 0) check.

@damienen

Comment (damienen) [reply src/common/mint-utils.ts:65]: Solved.

**Review by bucurdavid — APPROVED** [review]

**Comment by newbreedofgeek** [comment]
Looks good @damienen 

Once confirmed to work via an alpha pubish on NPM and used by Data DEX, we can merge to develop.

**Review by damienen — APPROVED** [review]

**Pull request merged**